internal/v1alpha2/source/image.go

```diff
@@ -0,0 +1,315 @@
+	// Validate inputs
+	if err := i.validate(source, opts); err != nil {
+		return nil, fmt.Errorf("validation unsuccessful during unpacking %v", err)
+	}
+	// storage path to store contents in local directory.
+	storagePath := filepath.Join(store.GetBundleDeploymentName(), filepath.Clean(source.Destination))
+	return i.unpack(ctx, storagePath, source, store, opts)
+}
+
+func (i *image) unpack(ctx context.Context, storagePath string, source *v1alpha2.BundleDeplopymentSource, store store.Store, opts UnpackOption) (*Result, error) {
+	pod := &corev1.Pod{}
+	op, err := i.ensureUnpackPod(ctx, store.GetBundleDeploymentName(), *source, pod, opts)
+	if err != nil {
+		return nil, err
+	} else if op == controllerutil.OperationResultCreated || op == controllerutil.OperationResultUpdated || pod.DeletionTimestamp != nil {
+		return &Result{State: StateUnpackPending}, nil
+	}
+	switch phase := pod.Status.Phase; phase {
+	case corev1.PodPending:
+		return pendingImagePodResult(pod), nil
+	case corev1.PodRunning:
+		return &Result{State: StateUnpacking}, nil
+	case corev1.PodFailed:
+		return nil, i.failedPodResult(ctx, pod)
+	case corev1.PodSucceeded:
+		return i.succeededPodResult(ctx, pod, storagePath, *source, store)
+	default:
+		return nil, i.handleUnexpectedPod(ctx, pod)
+	}
+}
+
+func (i *image) ensureUnpackPod(ctx context.Context, bdName string, bundleSrc v1alpha2.BundleDeplopymentSource, pod *corev1.Pod, opts UnpackOption) (controllerutil.OperationResult, error) {
+	// The name of the unpack pod is same as the bundle deployment. Since only one image source per bundle deployment
+	// is acceptable, this would not be an issue.
+	existingPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Namespace: i.PodNamespace, Name: bdName}}
+	if err := i.Client.Get(ctx, client.ObjectKeyFromObject(existingPod), existingPod); client.IgnoreNotFound(err) != nil {
+		return controllerutil.OperationResultNone, err
+	}
+
+	podApplyConfig := i.getDesiredPodApplyConfig(bdName, &bundleSrc, opts)
+	updatedPod, err := i.KubeClient.CoreV1().Pods(i.PodNamespace).Apply(ctx, podApplyConfig, metav1.ApplyOptions{Force: true, FieldManager: "rukpak-core"})
+	if err != nil {
+		if !apierrors.IsInvalid(err) {
+			return controllerutil.OperationResultNone, err
+		}
+		if err := i.Client.Delete(ctx, existingPod); err != nil {
+			return controllerutil.OperationResultNone, err
+		}
+		updatedPod, err = i.KubeClient.CoreV1().Pods(i.PodNamespace).Apply(ctx, podApplyConfig, metav1.ApplyOptions{Force: true, FieldManager: "rukpak-core"})
+		if err != nil {
+			return controllerutil.OperationResultNone, err
+		}
+	}
+
+	// make sure the passed in pod value is updated with the latest
+	// version of the pod
+	*pod = *updatedPod
+
+	// compare existingPod to newPod and return an appropriate
+	// OperatorResult value.
+	newPod := updatedPod.DeepCopy()
+	unsetNonComparedPodFields(existingPod, newPod)
```

Comment:
This is slightly hard to follow. If what you want is to compare `pod.Spec`, why not just do that?